Makefile

```diff
@@ -59,6 +59,24 @@ test-compile:
 	fi
 	go test -c $(TEST) $(TESTARGS)
 
+.PHONY: terraform
+TERRAFORM = ./bin/terraform
+terraform: ## Download terraform locally if necessary.
+ifeq (,$(wildcard $(TERRAFORM)))
+ifeq (,$(shell which $(notdir $(TERRAFORM)) 2>/dev/null))
+	@{ \
+	set -e ;\
+	mkdir -p $(dir $(TERRAFORM)) ;\
+	OS=$(shell go env GOOS) && ARCH=$(shell go env GOARCH) && \
+	curl -sfSLo $(TERRAFORM).zip https://releases.hashicorp.com/terraform/$(TERRAFORM_VERSION)/terraform_$(TERRAFORM_VERSION)_$${OS}_$${ARCH}.zip ; \
```

Comment:
Looks like the CI error we are getting is 404. I wonder if `OS` and `ARCH` are set here?